Source/Core/DolphinQt2/Settings/AudioPane.cpp

```diff
@@ -0,0 +1,229 @@
+// Copyright 2017 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include "DolphinQt2/Settings/AudioPane.h"
+
+#include <QCheckBox>
+#include <QComboBox>
+#include <QGridLayout>
+#include <QGroupBox>
+#include <QLabel>
+#include <QRadioButton>
+#include <QSlider>
+#include <QSpinBox>
+#include <QVBoxLayout>
+
+#include "AudioCommon/AudioCommon.h"
+#include "DolphinQt2/Config/SettingsWindow.h"
+#include "DolphinQt2/Settings.h"
+
+AudioPane::AudioPane()
+{
+  CreateWidgets();
+  LoadSettings();
+  ConnectWidgets();
+
+  connect(&Settings::Instance(), &Settings::VolumeChanged, this, &AudioPane::OnVolumeChanged);
+}
+
+void AudioPane::CreateWidgets()
+{
+  auto* dsp_box = new QGroupBox(tr("DSP Emulator Engine"));
+  auto* dsp_layout = new QVBoxLayout;
+  dsp_box->setLayout(dsp_layout);
+  m_dsp_hle = new QRadioButton(tr("DSP HLE emulation (fast)"));
+  m_dsp_lle = new QRadioButton(tr("DSP LLE recompiler"));
+  m_dsp_interpreter = new QRadioButton(tr("DSP LLE interpreter (slow)"));
+
+  dsp_layout->addWidget(m_dsp_hle);
+  dsp_layout->addWidget(m_dsp_lle);
+  dsp_layout->addWidget(m_dsp_interpreter);
+
+  auto* volume_box = new QGroupBox(tr("Volume"));
+  auto* volume_layout = new QGridLayout;
+  m_volume_slider = new QSlider;
+  m_volume_indicator = new QLabel(tr("??? %"));
```

Comment:
AFAICT the question marks are never supposed to appear, so why translate them? Can't you also set a blank label like `new QLabel()` and still make it work? Same comment for `m_stretching_buffer_indicator`